ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/engine/EmbeddedServer.kt

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.engine
+
+import io.ktor.events.*
+import io.ktor.server.application.*
+import io.ktor.util.logging.*
+import kotlinx.coroutines.*
+import kotlin.coroutines.*
+
+/**
+ * Represents an embedded server that hosts an application.
+ * It's an entry point to the application and handles the lifecycle of the application engine.
+ *
+ * @param TEngine The type of the application engine used by the server.
+ * @param TConfiguration The type of the configuration used by the engine.
+ */
+public expect class EmbeddedServer<TEngine : ApplicationEngine, TConfiguration : ApplicationEngine.Configuration>(
+    applicationProperties: ApplicationProperties,
+    engineFactory: ApplicationEngineFactory<TEngine, TConfiguration>,
+    engineConfigBlock: TConfiguration.() -> Unit = {}
+) {
+    /**
+     * Provides events on Application lifecycle
+     */
+    public val monitor: Events
+
+    public val environment: ApplicationEnvironment
+    public val application: Application
+    public val engine: TEngine
+    public val engineConfig: TConfiguration
+
+    public fun start(wait: Boolean = false): EmbeddedServer<TEngine, TConfiguration>
+
+    public fun stop(
+        gracePeriodMillis: Long = engineConfig.shutdownGracePeriod,
+        timeoutMillis: Long = engineConfig.shutdownGracePeriod
+    )
+}
+
+/**
+ * Factory interface for creating [ApplicationEngine] instances
+ */
+public interface ApplicationEngineFactory<
+    out TEngine : ApplicationEngine,
+    TConfiguration : ApplicationEngine.Configuration
+    > {
+
+    public fun configuration(configure: TConfiguration.() -> Unit): TConfiguration
+
+    /**
+     * Creates an engine from the given [environment] and [configure] script
+     */
+    public fun create(
+        environment: ApplicationEnvironment,
+        monitor: Events,
+        developmentMode: Boolean,
+        configuration: TConfiguration,
+        applicationProvider: () -> Application
+    ): TEngine
+}
+
+/**
+ * Creates an embedded server with the given [factory], listening on [host]:[port]
+ * @param watchPaths specifies path substrings that will be watched for automatic reloading
+ * @param configure configuration script for the engine
```

Comment:
I think `factory`,`host`, and `port` should be listed as params here and in the KDocs of other functions where they are used. 
Also, punctuation is missing at the end of the sentences.